pkg/tests/integration_test.go

```diff
@@ -33,8 +33,7 @@ func TestTypeCheckError(t *testing.T) {
 func TestMismatchedConfigType(t *testing.T) {
 	testWrapper(t, integrationDir("mismatched-config-type"), ExpectFailure, StderrValidator{
 		f: func(t *testing.T, stderr string) {
-			assert.Contains(t, stderr,
-				`config key "foo" cannot have conflicting types boolean, integer`)
+			assert.Regexp(t, `config key "foo" cannot have conflicting types boolean, (integer|number)`, stderr)
```

Comment:
Since PropertyValue's are always passed as floats we end up having a number type here instead of integer.  I don't think it makes much of a difference in practice, since this is all yaml anyway, but if I'm missing something we can probably try to cast it to integers.